### PR TITLE
add NVIDIA docker image

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

adding docker image based on `nvcr.io/nvidia/pytorch`
@edenlightning tested locally and works fine :]

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by s-rog — APPROVED** [review]
My env is a modified ngc pytorch image as well!

**Comment by SeanNaren** [comment]
Nice to hear @s-rog, the provided NGC images are awesome and I've been using them religiously for a while now. Comes with awesome niceties like MPI support (since they build torch from source!).

**Review by SeanNaren — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]
TBH I don't know much about docker but if @Borda says it works, it works. :smile:

**Review by ydcjeff — APPROVED** [review]

**Comment by Borda** [comment]
> TBH I don't know much about docker but if @Borda says it works, it works. 😄

add test for it in CI :]

**Review by awaelchli — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: .github/workflows/ci_dockers.yml [hunk 1]
```diff
@@ -150,3 +150,18 @@ jobs:
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      # https://github.com/docker/setup-buildx-action
+      # Set up Docker Buildx - to use cache-from and cache-to argument of buildx command
+      - uses: docker/setup-buildx-action@v1
+      - name: Build CUDA Docker
+        uses: docker/build-push-action@v2
+        with:
+          file: dockers/base-conda/Dockerfile
```
Comment (tchaton) [inline .github/workflows/ci_dockers.yml:165]: Shouldn't it be `dockers/nvidia/Dockerfile` ?

Comment (Borda) [reply .github/workflows/ci_dockers.yml]: good catch, fixed

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**